streams/src/main/java/org/apache/kafka/streams/processor/StateStore.java

```diff
@@ -20,13 +20,17 @@
 
 /**
  * A storage engine for managing state maintained by a stream processor.
- *
+ * <p>
+ * If the store is implemented as a persistent store, it <em>must</em> use the store name as directory to write to disk.
+ * Thus, the directory for the store is {@code <state.dir>/<application.id>/<taskId>/<storeName>}.
```

Comment:
Do we want to expose the pattern of `<state.dir>/<application.id>/<taskId>`? Could we just say users must create the directory with the name equal to the state store name under the directory given by `ProcessorContext#stateDir()`?

Reply:
I agree with @guozhangwang on this, as if we change our approach we'd have to re-educate users on the naming convention.